src/main/java/org/jabref/logic/citationstyle/CitationStyle.java

```diff
@@ -237,4 +238,23 @@ public boolean isInternalStyle() {
     public String getPath() {
         return getFilePath();
     }
+
+    public boolean isNumericStyle() {
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
Document Builder is costly to be initalied, make it a static const     private static final DocumentBuilderFactory DOCUMENT_BUILDER_FACTORY = DocumentBuilderFactory.newInstance();

Reply:
Ideally, this should be in the citeproc java

Reply:
TODO: File an upstream issue for suggesting a method for like isNumericStyle

Reply:
Since https://github.com/JabRef/jabref/pull/11604 is merged, we now have an xml builder initialized. How about I use stax parsing here as well?

Reply:
resolved